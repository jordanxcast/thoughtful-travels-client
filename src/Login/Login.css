.Login{
  margin-top: 150px;
  border: 2px solid #131B23;
  border-radius: 20px;
  padding: 30px;
  display: flex;
  flex-direction: column;
  margin: 100px 10%;
}

.Login-submit {
  align-self: center;
  padding: 10px 20px;
  font-size: 16px;
  border: 2px solid #627C85;
  border-radius: 20px;
  margin-top: 30px;
}

.Login-submit:hover {
  cursor: pointer;
  border-color: #808F85;
}