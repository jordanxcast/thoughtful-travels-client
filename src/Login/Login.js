import React, { Component } from 'react'
import TokenService from '../services/token-service'
import AuthApiService from '../services/auth-api-services'
import ApiContext from '../ApiContext'

export default class Login extends Component {
static contextType = ApiContext


handleSubmitBasicAuth = e => {
  e.preventDefault()
  const { user_name, password } =e.target
  
  TokenService.saveAuthToken(
    TokenService.makeBasicAuthToken(user_name.value, password.value)
  )

  user_name.value=''
  password.value=''
}

handleSubmitJwtAuth = e => {
  e.preventDefault()
  this.setState({ error: null })
  const { user_name, password } = e.target

  AuthApiService.postLogin({
    username: user_name.value,
    password: password.value,
  })
    .then(res => {
      console.log(res)
      user_name.value = ''
      password.value = ''
      TokenService.saveAuthToken(res.authToken)
      this.context.handleAuthToken(res.authToken)
      this.props.history.push(`/destinations`)
    })
    .catch(res => {
      this.setState({ error: res.error })
    })

}

render() {

  return (
    <form
      className='Login'
      onSubmit={this.handleSubmitJwtAuth}
    >
      <div className='user_name'>
        <label htmlFor='Login-user-name'>
          User name
        </label>
        <input
          name='user_name'
          id='Login-user-name'>
        </input>
      </div>
      <div className='password'>
        <label htmlFor='Login-password'>
          Password
        </label>
        <input
          name='password'
          type='password'
          id='Login-password'
        >
        </input>
        <button type='submit'>
          Login
        </button>
      </div>

    </form>
  )
}
}