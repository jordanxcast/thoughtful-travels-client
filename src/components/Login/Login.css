.Login-container {
  background-color: #E9EBF8;
  height: 100vh;
  margin: 0;
}

.Login{
  position: relative;
  top: 100px;
  margin-top: 40px;
  border: 2px solid #131B23;
  border-radius: 20px;
  padding: 30px;
  display: flex;
  flex-direction: column;
  margin: 0 10%;
  box-shadow: 0px 0px 5px 6px rgba(0,0,0,0.7);
  background-color: #B3BFB8;
  color: #131B23;
}

.Login > div  {
  display: flex;
  flex-direction: column;
  margin-top: 30px;
}

.password > input,
.user_name >input {
  padding: 5px;
  border: 1px solid black;
  border-radius: 10px;
}

.Login-submit {
  align-self: center;
  padding: 10px 20px;
  font-size: 16px;
  border: 2px solid #627C85;
  border-radius: 20px;
  margin-top: 30px;
}

.Login-submit:hover {
  cursor: pointer;
  border-color: #808F85;
}

@media (min-width: 500px) {
  .Login > div {
    display: flex;
    flex-direction: row;
    width: fit-content;
    margin: 20px auto;
    align-self: center;
  }
}