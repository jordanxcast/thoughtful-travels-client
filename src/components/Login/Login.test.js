import React from 'react'
import ReactDOM from 'react-dom'
import Login from './Login'
import renderer from 'react-test-renderer';


describe(`<Login />`, () => {
  //Smoke Testing
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Login 
        
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div)
  })
  //Snapshot Testing
  it('renders UI as expected', () => {
    const tree = renderer
      .create(<Login />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})