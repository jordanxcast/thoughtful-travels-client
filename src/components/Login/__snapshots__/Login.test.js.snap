// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders UI as expected 1`] = `
<div
  className="Login-container"
>
  <form
    className="Login"
    onSubmit={[Function]}
  >
    <h1>
      Login to Begin your Journey
    </h1>
    <div
      className="user_name"
    >
      <label
        htmlFor="Login-user-name"
      >
        *User name
      </label>
      <input
        id="Login-user-name"
        name="user_name"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="Login-password"
      >
        *Password
      </label>
      <input
        id="Login-password"
        name="password"
        type="password"
      />
    </div>
    <button
      className="Login-submit"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
